src/Classes/ServiceAPI/MyRadio_ShortURL.php

```diff
@@ -0,0 +1,196 @@
+                RETURNING short_url_id';
+        $result = self::$db->fetchOne($sql, [$slug, $redirectTo]);
+
+        return self::getInstance($result['short_url_id']);
+    }
+
+    public function getID()
+    {
+        return $this->short_url_id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSlug(): string
+    {
+        return $this->slug;
+    }
+
+    /**
+     * @param string $slug
+     * @return MyRadio_ShortURL
+     */
+    public function setSlug(string $slug): MyRadio_ShortURL
+    {
+        self::$db->query(
+            'UPDATE public.short_urls
+            SET slug = $2
+            WHERE short_url_id = $1',
+            [$this->short_url_id, $slug]
+        );
+        $this->slug = $slug;
+        $this->updateCacheObject();
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getRedirectTo(): string
+    {
+        return $this->redirect_to;
+    }
+
+    /**
+     * @param string $redirectTo
+     * @return MyRadio_ShortURL
+     */
+    public function setRedirectTo(string $redirectTo): MyRadio_ShortURL
+    {
+        self::$db->query(
+            'UPDATE public.short_urls
+            SET redirect_to = $2
+            WHERE short_url_id = $1',
+            [$this->short_url_id, $redirectTo]
+        );
+        $this->slug = $redirectTo;
+        $this->updateCacheObject();
+        return $this;
+    }
+
+    /**
+     * Deletes this short URL.
+     */
+    public function delete()
+    {
+        self::$db->query(
+            'DELETE FROM public.short_urls
+            WHERE short_url_id = $1',
+            [$this->short_url_id]
+        );
+    }
+
+    /**
+     * @return MyRadio_ShortURL[]
+     */
+    public static function getAll()
+    {
+        $rows = self::$db->fetchColumn(
+            'SELECT short_url_id FROM public.short_urls',
+            []
+        );
+        $results = [];
+        foreach ($rows as $id) {
+            $results[] = self::getInstance($id);
+        }
+        return $results;
+    }
+
+    public function logClick($userAgent, $ipAddress)
+    {
+        self::$db->query(
+            'INSERT INTO public.short_url_clicks (short_url_id, click_time, user_agent, ip_address)
+            VALUES ($1, NOW(), $2, $3)',
+            [$this->short_url_id, $userAgent, $ipAddress]
+        );
+    }
+
+    public static function getForm(): MyRadioForm
+    {
+        $domain = preg_replace('{^//}', '', Config::$website_url);
+        return (new MyRadioForm(
+            'shorturlfrm',
+            'Website',
+            'editShortUrl',
+            [
+                'title' => 'Edit Short URL'
+            ]
+        ))->addField(
+            new MyRadioFormField(
+                'slug',
+                MyRadioFormField::TYPE_TEXT,
+                [
+                    'label' => 'Slug',
+                    'explanation' => "The bit that goes after https://$domain. Don't include $domain or a slash."
```

Comment:
does this/could this fill from the config `short_url` or whatever the fields called?

Reply:
I don't think we have one of those - might you be confusing it with `short_name`?

Reply:
uh.. we've got `base_url` but thats got the /myradio bit on.